pentaho/impl/src/main/java/pt/webdetails/cdf/dd/DashboardDesignerContentGenerator.java

```diff
@@ -134,6 +141,32 @@ public String getObjectName() {
     return this.getClass().getName();
   }
 
+  private void translateResponseToServletResponse( IParameterProvider pathParams ) throws IOException {
+    // This is necessary when we need to convert the ResourcesApi Response object into the HttpServletResponse object
+    // we are using in this class.
+    HttpServletResponse servletResponse = getResponse();
+    ResourcesApi resourcesApi = new ResourcesApi();
+    String resourceParam = pathParams.getStringParameter( MethodParams.COMMAND, "" );
+    // First retrieve the ResourcesApi Response object
+    Response resourceResponse = resourcesApi.getResource( resourceParam, null );
+
+    // We should check to see if the status is OK first before continuing. Also need to send over the status as well
+    servletResponse.setStatus( resourceResponse.getStatus() );
+    if ( resourceResponse.getStatus() == Response.Status.OK.getStatusCode() ) {
+      // Next convert the headers over (the headers from resourceResponse are OutBoundHeaders which use a String-key
+      // to LinkedList-values), while HttpServletResponse uses more of a flat list type, so need to convert them.
+      resourceResponse.getMetadata().forEach( (key, value) ->
+        value.forEach( headerValue -> servletResponse.setHeader( key, (String) headerValue ) ) );
+
+      // Unable to safely retrieve the contents of the resourceResponse's output stream (i.e. JSON translated properly,
```

Comment:
I have had issues with similar methods of reading the entity, I will try your approach to see if I can get the data, I agree that if we can get it once instead of calling that public method (then we can remove the public method altogether). That would be the ideal scenario, my previous methods worked for json but not png.  Will look into this too

Reply:
I remember now, when debugging and testing this issue, readEntity is not seen as a method and this is due to the libraries we have deployed in the server.  We use the Abstract Response object from jsr311-api and it is implemented by jersey-core.  The IDE for some reason is suggesting using readEntity, which is used by javax.ws.rs-api.  javax.ws.rs-api is the official spec jar for JAX-RS 2.x while jsr311-api is the spec jar for JAX-RS 1.x.  We are not deploying the implementation jar for JAX-RS 2.x (jaxrs-ri.jar) that would allow us to use the readEntity method.
I believe this is an issue with upgrading our JAX to the new implementation and it has not been done yet, as such, we cannot use readEntity until such upgrade occurs.
(reference: https://stackoverflow.com/questions/32106428/jaxrs-api-vs-jsr311-api-vs-javax-ws-rs-api-vs-jersey-core-vs-jaxrs-ri)

Reply:
Your point gave me an idea, it looks like this code works:

((StreamingOutput) resourceResponse.getEntity()).write( servletResponse.getOutputStream() );

I'm getting a positive on JSON and PNG files. So that should clean up some unnecessary changes, thanks @dcleao